presto-test-jdbc-compatibility-old-server/src/test/java/io/prestosql/TestJdbcResultSetCompatibility.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logging;
+import io.prestosql.jdbc.BaseTestJdbcResultSet;
+import io.prestosql.jdbc.PrestoDriver;
+import org.testcontainers.containers.PrestoContainer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true)
+public class TestJdbcResultSetCompatibility
+        extends BaseTestJdbcResultSet
+{
+    private static final int NUMBER_OF_TESTED_VERSIONS = 5;
+    private static final int TESTED_VERSIONS_GRANULARITY = 3;
+
+    private final Optional<String> testedPrestoVersion;
```

Comment:
```suggestion
    /**
     * Test version. Empty means current (unreleased) version
     */
    private final Optional<String> testedPrestoVersion;
```

Reply:
It means we could not obtain presto version. And therefore tests defined here will fail.
I will update comment to reflect to that.

(I cannot fail directly in `testedPrestoVersions` data provider as it kills whole testNG)

Reply:
> (I cannot fail directly in testedPrestoVersions data provider as it kills whole testNG)

if this results in surefire's failure, that's fine to me